File: src/main/java/com/likelionknu/applyserver/mail/data/repository/MailHistoryRepository.java
package com.likelionknu.applyserver.mail.data.repository;

import com.likelionknu.applyserver.auth.data.entity.User;
import com.likelionknu.applyserver.mail.data.entity.MailHistory;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface MailHistoryRepository extends JpaRepository<MailHistory, Long> {
    List<MailHistory> findAllByUser(User user);
}

File: src/main/kotlin/com/likelionknu/applyserver/application/data/repository/MailHistoryRepository.kt
package com.likelionknu.applyserver.application.data.repository

import com.likelionknu.applyserver.application.data.entity.MailHistory
import com.likelionknu.applyserver.auth.data.entity.User
import org.springframework.data.jpa.repository.JpaRepository

interface MailHistoryRepository : JpaRepository<MailHistory, Long> {
    fun findAllByUser(user: User): List<MailHistory>
}